webdriver/search_test.go

```diff
@@ -24,58 +32,62 @@ func TestSearch(t *testing.T) {
 	defer service.Stop()
 	defer wd.Quit()
 
-	testSearch(t, wd, app, "/", "wpt-results")
-	testSearch(t, wd, app, "/interop/", "wpt-interop")
-}
-
-func testSearch(t *testing.T, wd selenium.WebDriver, app AppServer, path string, elementName string) {
 	// Navigate to the wpt.fyi homepage.
 	if err := wd.Get(app.GetWebappURL(path)); err != nil {
 		panic(err)
 	}
 
 	// Wait for the results view to load.
-	numInitialPathParts := 0
-	runsLoadedCondition := func(wd selenium.WebDriver) (bool, error) {
+	resultsLoadedCondition := func(wd selenium.WebDriver) (bool, error) {
 		pathParts, err := getPathPartElements(wd, elementName)
 		if err != nil {
 			return false, err
 		}
-		numInitialPathParts = len(pathParts)
 		return len(pathParts) > 0, nil
 	}
-	err := wd.WaitWithTimeout(runsLoadedCondition, time.Second*10)
+	err = wd.WaitWithTimeout(resultsLoadedCondition, time.Second*5)
 	assert.Nil(t, err)
 
 	// Run the search.
 	searchBox, err := getSearchElement(wd, elementName)
 	if err != nil {
 		panic(err)
 	}
-	const query = "2dcontext"
-	if err := searchBox.SendKeys(query + selenium.EnterKey); err != nil {
+	folder := "2dcontext"
+	if err := searchBox.SendKeys(folder + selenium.EnterKey); err != nil {
 		panic(err)
 	}
+	assertListIsFiltered(t, wd, elementName, folder)
+
+	// Navigate to the wpt.fyi homepage.
+	if err := wd.Get(app.GetWebappURL(path) + "?q=" + folder); err != nil {
+		panic(err)
+	}
+	err = wd.WaitWithTimeout(resultsLoadedCondition, time.Second*5)
+	assert.Nil(t, err)
+	assertListIsFiltered(t, wd, elementName, folder)
+}
+
+func assertListIsFiltered(t *testing.T, wd selenium.WebDriver, elementName, folder string) {
+	var pathParts []selenium.WebElement
+	var err error
 	filteredPathPartsCondition := func(wd selenium.WebDriver) (bool, error) {
-		pathParts, err := getPathPartElements(wd, elementName)
+		pathParts, err = getPathPartElements(wd, elementName)
 		if err != nil {
 			return false, err
 		}
-		return len(pathParts) > 0 && len(pathParts) < numInitialPathParts, nil
+		return len(pathParts) == 1, nil
```

Comment:
Does this test use static input data? If not, the introduction of something in another folder named `2dcontext` could cause an unexpected breakage.

Reply:
It does locally, but does not in staging. Been this way a long time.. I'll leave it for you to have the chance to say "told you so" later ;)